base/strings/string.jl

```diff
@@ -172,15 +172,13 @@ end
 
 ## checking UTF-8 & ACSII validity ##
 
-byte_string_classify(data::Vector{UInt8}) =
-    ccall(:u8_isvalid, Int32, (Ptr{UInt8}, Int), data, length(data))
-byte_string_classify(s::String) =
+byte_string_classify(s::Union{String,SubString{String},Vector{UInt8}}) =
     ccall(:u8_isvalid, Int32, (Ptr{UInt8}, Int), s, sizeof(s))
     # 0: neither valid ASCII nor UTF-8
     # 1: valid ASCII
     # 2: valid UTF-8
 
-isvalid(::Type{String}, s::Union{Vector{UInt8},String}) = byte_string_classify(s) != 0
+isvalid(::Type{String}, s::Union{Vector{UInt8},String,SubString{String}}) = byte_string_classify(s) != 0
```

Comment:
@Vaibhavdixit02 I would recommend to add `isvalid(<:SubString{String})` definition and update an appropriate docstring.

Reply:
Why is that required? I am not familiar with any subtype of `SubString{String}`, are there any?

Reply:
You are right - there are none. I meant that single argument `isvalid` should be defined as:
```
isvalid(s::Union{String,SubString{String}}) = isvalid(String, s)
```

Reply:
and that this change should be also reflected in docstrings

Reply:
Ah! Got it.

Reply:
Why has it not been extended for `Vector{UInt8}` ?

Reply:
Sure

Reply:
I meant that in current implementation it seems that without explicitly telling `String` even `Vector{UInt8}` returns `false` so is this extension something that would be added or is it not in line with `isvalid()`'s required features? @stevengj  perhaps you could give your view too?

Reply:
I meant that `isvalid(::Vector{UInt8})` would now throw method error and since `isvalid(::Type{String}, ::Vector{UInt8})` is allowed I would also allow it.

Reply:
`isvalid(::Vector{UInt8})` would be really confusing. There's way more things `Vector{UInt8}` can be other than a string.

Reply:
`isvalid(SubString{String})` should work though, right? @yuyichao

Reply:
`isvalid(String, ::Vector{UInt8})` should continue to work.  I agree that `isvalid(::Vector{UInt8})` makes no sense.